can-deparam.js

```diff
@@ -42,6 +42,9 @@ var digitTest = /^\d+$/,
 	paramTest = /([^?#]*)(#.*)?$/,
 	entityRegex = /%([^0-9a-f][0-9a-f]|[0-9a-f][^0-9a-f]|[^0-9a-f][^0-9a-f])/i,
 	prep = function (str) {
+		if (str.charAt(0) === "#" || str.charAt(0) === "?") {
```

Comment:
Another way to write this might be ...

between `entityRegex ` and `prep` create:

```
startChars = {"#": true,"?": true}
``

Then use that like this:

```
if (startChars[str.charAt(0)] === true) {
  str = str.substr(1); 
}
```

The 2nd argument isn't needed